### PR TITLE
Enable pytest-shard for functorch

### PR DESCRIPTION
**Pull request by huydhn**
This extends https://github.com/pytorch/pytorch/pull/84961 to support functorch tests with pytest-shard.

### CODE REVIEW

File: functorch/test/conftest.py [hunk 1]
```diff
@@ -0,0 +1,146 @@
+from _pytest.junitxml import LogXML, _NodeReporter, bin_xml_escape
```
Comment (samdow) [inline functorch/test/conftest.py:1]: Just to check, this is a straight copy of test/conftest.py to make functorch work even though its tests aren't in `test/`?

Comment (huydhn) [reply functorch/test/conftest.py:1]: Yeah, exactly. This is the hack that I mention. This file extends pytest capability a bit and it needs to be there on functorch test directory AFAIK

**Review by samdow — APPROVED** [review]
LGTM from the functorch side (i.e. skimmed that all the functorch tests look to be running), but I definitely don't know the build stuff as well so might want to have someone else look at that. Thanks for doing this!

**Comment by huydhn** [comment]
> LGTM from the functorch side (i.e. skimmed that all the functorch tests look to be running), but I definitely don't know the build stuff as well so might want to have someone else look at that. Thanks for doing this!

Yeah, I'm asking folks on Dev Infra to look into this too before marking this as done.

**Review by clee2000 — APPROVED** [review]

**Pull request closed without merging**